plugins/woocommerce/tests/php/includes/class-wc-checkout-test.php

```diff
@@ -97,6 +97,34 @@ public function test_order_notes() {
 		$this->assertStringContainsString( 'This text should not save inside an anchor.', $content );
 	}
 
+	/**
+	 * @testdox the customer notes can have linebreaks.
+	 */
+	public function test_order_notes_linebreaks() {
+		$data = array(
+			'ship_to_different_address' => false,
+			'order_comments'            => 'A string
+			with linebreaks
+			in it.',
+			'payment_method'            => WC_Gateway_BACS::ID,
+		);
+
+		$errors = new WP_Error();
+
+		$this->sut->validate_posted_data( $data, $errors );
+		$result = $this->sut->create_order( $data );
+
+		$content = wc_get_template_html(
+			'order/order-details.php',
+			array(
+				'order_id'       => $result,
+				'show_downloads' => false,
+			)
+		);
+		// The preg_replace is necessary because the template outputs a lot of whitespace, we can just make sure the <br /> tags are there as the other whitespace doesn't matter.
+		$this->assertStringContainsString( 'A string<br />with linebreaks<br />in it.', preg_replace( '/[\t|\\n]+/', '', $content ) );
```

Comment:
It's not removing the newline but the literal `\n` substring, we can add the carriage return `\r`, and there's no need for `|` there (this is taken as literal inside the brackets, so it is replacing the `|`) 👍🏼 

```suggestion
		$this->assertStringContainsString( 'A string<br />with linebreaks<br />in it.', preg_replace( '/[\t\n\r]+/', '', $content ) );
```

Reply:
The love I feel for this tool [https://www.phpliveregex.com/](https://www.phpliveregex.com/#tab-preg-replace) is outlasting my relationships 😂